pyccl/halos/profiles.py

```diff
@@ -267,17 +275,167 @@ def cumul2d(self, cosmo, r_t, M, a, mass_def=None):
+        Dls = angular_diameter_distance(cosmo, a_lens, a_src)
+        A = (
+            physical_constants.CLIGHT ** 2
+            * physical_constants.MPC_TO_METER
+            / (4.0 * np.pi * physical_constants.GNEWT
+               * physical_constants.SOLAR_MASS)
+        )
+
+        Sigma_crit = A * Ds / (Dl * Dls)
+        return Sigma_crit
+
+    def convergence(self, cosmo, r, M, a_lens, a_src, mass_def=None):
+        """ Returns the convergence as a function of cosmology,
+        radius, halo mass and the scale factors of the source
+        and the lens.
+
+        .. math::
+           \\kappa(R) = \\frac{\\Sigma(R)}{\\Sigma_{\\mathrm{crit}}},\\
+
+        where :math:`\\Sigma(R)` is the 2D projected surface mass density.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
+            r (float or array_like): comoving radius in Mpc.
+            M (float or array_like): halo mass in umits of M_sun.
+            a_lens (float or array_like): lens' scale factor.
+            a_src (float or array_like): source's scale factor.
+            mass_def (:class:`~pyccl.halos.massdef.MassDef`):
+                a mass definition object.
+
+        Returns:
+            float or array_like: convergence \
+                :math:`\\kappa`
+        """
+
+        Sigma = self.projected(cosmo, r, M, a_lens, mass_def) / a_lens**2
+        Sigma_crit = self.sigma_critical(cosmo, a_lens, a_src)
+        return Sigma / Sigma_crit
+
+    def shear(self, cosmo, r, M, a_lens, a_src, mass_def=None):
+        """ Returns the shear (tangential) as a function of cosmology,
+        radius, halo mass and the scale factors of the
+        source and the lens.
+
+        .. math::
+           \\gamma(R) = \\frac{\\Delta\\Sigma(R)}{\\Sigma_{\\mathrm{crit}}} =
+           \\frac{\\overline{\\Sigma}(< R) -
+           \\Sigma(R)}{\\Sigma_{\\mathrm{crit}}},\\
+
+        where :math:`\\overline{\\Sigma}(< R)` is the average surface density
+        within R.
+
+        Args:
+            cosmo (:class:`~pyccl.core.Cosmology`): A Cosmology object.
+            r (float or array_like): comoving radius in Mpc.
+            M (float or array_like): halo mass in umits of M_sun.
+            a_lens (float or array_like): lens' scale factor.
+            a_src (float or array_like): source's scale factor.
+            mass_def (:class:`~pyccl.halos.massdef.MassDef`):
+                a mass definition object.
+
+        Returns:
+            float or array_like: shear \
+                :math:`\\gamma`
+        """
+
+        Sigma = self.projected(cosmo, r, M, a_lens, mass_def)
+        Sigma_bar = self.cumul2d(cosmo, r, M, a_lens, mass_def)
+        Sigma_crit = self.sigma_critical(cosmo, a_lens, a_src)
+        return (Sigma_bar - Sigma) / (Sigma_crit * a_lens**2)
+
+    def reduced_shear(self, cosmo, r, M, a_lens, a_src, mass_def=None):
+        """ Returns the reduced shear as a function of cosmology,
```

Comment:
In most of the code above, the space after the opening quotations for the docstrings has been removed. However, in this class, it has been reinstated.